File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number}
 */
var maxDepth = function(root) {
    
    
    return traverse(root, 0)
    
    function traverse(current, level){
        if(!current){
            return level
        }
        
        let leftLevel=0|| Math.max(leftLevel,traverse(current.left, level+1))
        let rightLevel=0|| Math.max(rightLevel,traverse(current.right, level+1))
        
        return Math.max(leftLevel, rightLevel)
        
    }
       
    
};

File: 0104-maximum-depth-of-binary-tree/0104-maximum-depth-of-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxDepth(self, root: Optional[TreeNode]) -> int:
        answer = 0
        maxLeft = 0
        answer = self.dfs(root, 0)

        return answer

    def dfs(self, node, level):
        if node is None:
            return level
        level += 1

        maxL = self.dfs(node.left, level)
        maxR = self.dfs(node.right, level)

        maxDepth = max(maxL, maxR)

        return maxDepth
